### PR TITLE
Upgrade to PDK 2.5.0

### PR DESCRIPTION
**Pull request by Phil-Friderici**
- uses rspec-puppet-facts for testing
- drops support for stringified bools and integers
- drops support for Puppet 3

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -1,135 +1,136 @@
...
-  $hiera_merge_sudoers       = false,
+  String[1]                                      $package                   = 'sudo',
+  Optional[String[1]]                            $package_source            = undef,
+  Enum['present','absent']                       $package_ensure            = 'present',
+  Boolean                                        $package_manage            = true,
+  Optional[Stdlib::Absolutepath]                 $package_adminfile         = undef,
+  Stdlib::Absolutepath                           $config_dir                = '/etc/sudoers.d',
+  String[1]                                      $config_dir_group          = 'root',
+  Stdlib::Filemode                               $config_dir_mode           = '0750',
+  Enum['present', 'absent', 'directory', 'link'] $config_dir_ensure         = 'directory',
```
Comment (anders-larsson) [inline manifests/init.pp:108]: We do not allow setting the file resource's target. Maybe there is no point in allowing `link`?

File: manifests/init.pp [hunk 2]
```diff
@@ -1,135 +1,136 @@
...
+  Boolean                                        $envreset                  = true,
+  Array                                          $envkeep                   = ['COLORS','DISPLAY','HOSTNAME','HISTSIZE','INPUTRC','KDEDIR','LS_COLORS','MAIL','PS1','PS2','QTDIR','USERNAME','LANG','LC_ADDRESS','LC_CTYPE','LC_COLLATE','LC_IDENTIFICATION','LC_MEASUREMENT','LC_MESSAGES','LC_MONETARY','LC_NAME','LC_NUMERIC','LC_PAPER','LC_TELEPHONE','LC_TIME','LC_ALL','LANGUAGE','LINGUAS','_XKB_CHARSET','XAUTHORITY'], #lint:ignore:140chars
+  String[1]                                      $secure_path               = '/sbin:/bin:/usr/sbin:/usr/bin',
+  Boolean                                        $root_allow_all            = true,
+  Boolean                                        $includedir                = true,
+  Boolean                                        $include_libsudo_vas       = false,
+  Stdlib::Absolutepath                           $libsudo_vas_location      = '/opt/quest/lib64/libsudo_vas.so',
+  Optional[Boolean]                              $always_query_group_plugin = undef,
+  Boolean                                        $hiera_merge_sudoers       = false,
```
Comment (anders-larsson) [inline manifests/init.pp:127]: Should we opt to do as we have done in other modules and scrap "merge" parameters and instead rely on native merge in Hiera itself? Might be suitable to keep it in its own commit (or maybe pull request though)?

**Review by anders-larsson — COMMENTED** [review]
Looks good. However gave some suggestions.

File: manifests/init.pp (hunk 1 above)
Comment (Phil-Friderici) [reply manifests/init.pp:108]: I'll remove 'link' then.

File: manifests/init.pp (hunk 2 above)
Comment (Phil-Friderici) [reply manifests/init.pp:127]: Yes sure, removing hiera_merge will ease testing quite a lot :)
Let me remove $hiera_merge_sudoers and the according tests too.

File: manifests/init.pp (hunk 1 above)
Comment (Phil-Friderici) [reply manifests/init.pp]: removed with commit [4079f70](https://github.com/Ericsson/puppet-module-sudo/pull/39/commits/4079f703a4e6c59e4a343294007acb9acdd3d313)

File: manifests/init.pp (hunk 2 above)
Comment (Phil-Friderici) [reply manifests/init.pp]: removed with commit [f86c8b5](https://github.com/Ericsson/puppet-module-sudo/pull/39/commits/f86c8b5c6fad97cdcf24c524e93edd791ce2a278)

**Comment by Phil-Friderici** [comment]
Anything else to do before merging ?

**Review by anders-larsson — APPROVED** [review]

**Comment by Phil-Friderici** [comment]
Released as v1.0.0
